lib/ocpp/v201/charge_point.cpp

```diff
@@ -33,15 +33,30 @@ bool Callbacks::all_callbacks_valid() const {
            this->connector_effective_operative_status_changed_callback != nullptr and
            this->get_log_request_callback != nullptr and this->unlock_connector_callback != nullptr and
            this->remote_start_transaction_callback != nullptr and this->is_reservation_for_token_callback != nullptr and
-           this->update_firmware_request_callback != nullptr and
+           this->update_firmware_request_callback != nullptr and this->security_event_callback != nullptr and
+           this->set_charging_profiles_callback != nullptr and
```

Comment:
@shankari: The `security_event_callback` is defined as non-optional in `charge_point.hpp`:

https://github.com/EVerest/libocpp/blob/77f5b2a6181d94c2e94af790e4c5e61435f0b919/include/ocpp/v201/charge_point.hpp#L146-L152

It looks to me like @gberardi-pillar's change is meant to correct for this.

Reply:
That said, the only usage of `security_event_callback` in `libocpp` appears to assume that it's optional:

https://github.com/EVerest/libocpp/blame/587d491589f944c8a99ce8fe56ece612e061f385/lib/ocpp/v201/charge_point.cpp#L1878-L1880

So perhaps it would be better to change the type of `security_event_callback` to `std::optional<std::function>` instead of add it to the validate method?

Reply:
I have added these comments.